src/main/java/org/opentest4j/fixtures/FixtureContext.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j.fixtures;
+
+/**
+ * Provides methods to register {@link TearDown} and {@link TestPostcondition} instances.
+ *
+ * @author Kevin Cooney
+ * @since 1.1
+ */
+public abstract class FixtureContext {
+
+	/**
+	 * Adds a {@link TearDown} to run after the test completes.
+	 */
+	public abstract void addTearDown(TearDown tearDown);
+
+	/**
+	 * Adds a {@link TestPostcondition} to run after a successful test run. The
+	 * post-condition will run before any {@link TearDown} instances, but
+	 * may not run at all if an easier post-condition fails.
+	 */
+	public abstract void addTestPostcondition(TestPostcondition postcondition);
+
+	/**
+	 * Gets the method that this fixture is modifying, or {@code null} if this
+	 * fixture is a class fixture.
+	 */
+	public abstract InstanceMethod getInstanceMethod();
+
+	/**
+	 * Gets the class that this fixture is modifying.
+	 */
+	public abstract ClassWrapper getTestClass();
```

Comment:
I think both test class and instance method should be optional, i.e. allow null to be returned. If a fixture cannot work with that it should be allowed to throw an exception.